Updating parsers/tests, fixing phpstan error, adding automatic test

jaydiablo opened the pull request:
Testing to see if I can get travis to run one of the parse commands against all parsers to make sure updates to lock files and such don't break any parsing.

jaydiablo committed:
Updating parsers/tests, fixing phpstan error, adding automatic test

jaydiablo committed:
fixing test command

mimmi20 commented on src/Command/Compare.php:
```diff
@@ -16,8 +18,12 @@ protected function configure(): void
             ->setDescription('Runs tests, normalizes the results then analyzes the results')
-            ->addArgument('run', InputArgument::OPTIONAL, 'The name of the test run, if omitted will be generated from date')
-            ->addArgument('file', InputArgument::OPTIONAL, 'Path to a file to use as the source of useragents rather than test suites')
+            ->setDefinition(
```
Could you change the other commands in the same way too?

jaydiablo replied:
Do you mean using `setDefinition` or changing from Argument to Option?

The reason I changed this one was because it wasn't possible to set just the `file` argument since it was expecting `./bin/console compare <run> <file>`. Now the `file` argument can be specified right after the command, and then `run` option uses a parameter prefix: `./bin/console compare <file> --run=<run>`.

I don't think the other commands have that issue, but I can check.

As for the syntax, I just changed that because it was the example used in the Symfony Console documentation: https://symfony.com/doc/current/components/console/console_arguments.html  I think either syntax is acceptable (the `addArgument` and `addOption` syntax is just short-hand).

mimmi20 replied:
I mean both changes. For me it's easier to understand what commands do, when they are defined in a similar way.

jaydiablo committed:
calling prepare instead of update, fixing php-get-browser update command if no version.txt file

jaydiablo committed:
changing from shrinkwrap to package-lock

This isn't really necessary, just updating since I was in there. Not sure why this parser fails on travis.

jaydiablo committed:
trying specifying node version, updating NPM in before_install

jaydiablo committed:
Updating parser lock files

Mainly to see if device-detector-js-1 has been fixed.

jaydiablo committed:
Updating parser/test lockfiles again, reverting the definition change for Compare command to short-hand style

psalm-shepherd commented:
Psalm didn’t find any errors!

jaydiablo commented:
I think I'd still like to adjust the return code of the Compare command so that it sends a non-zero code when a parser fails to parse (right now there's one failing, I assume due to NodeJS version in travis. All parsers work fine on my machine).  I think that can be added later though, don't want the build to fail because of this just yet (need to address that broken parser on travis).

mimmi20 commented:
Could you add some information to this travis test in the future? Like parser versions, used version of PHP and Node.js?

This would be helpful to create issues in the parser projects.

jaydiablo commented:
@mimmi20 what do you mean? In the output of the test run?

mimmi20 commented:
@jaydiablo You mentioned this:

> (right now there's one failing, I assume due to NodeJS version in travis

I looked into the run and wanted to create an issue in that library. But its actually not possible to say which version of the parser, PHP and node are used, because this information is missing in the output.

To have this kind of information would be helpful.

jaydiablo commented:
Ah, yeah, could probably make that more obvious, the information does typically show in the travis build log, but can be sort of hidden:

PHP version: https://travis-ci.org/diablomedia/useragent-parser-comparison/jobs/562780418#L466 (7.2.16)
NPM version: https://travis-ci.org/diablomedia/useragent-parser-comparison/jobs/562780418#L487 (5.5.1)
Node version: https://travis-ci.org/diablomedia/useragent-parser-comparison/jobs/562780418#L479 (8.9.1)

The travis.yml file does specify Node 8 (added in this PR) and PHP 7.2 as the php version it'll run as, but those are just the Major versions.

jaydiablo committed:
Merge branch 'master' into auto-test

psalm-shepherd commented:
Psalm didn’t find any errors!
